scanners/web-endpoint-puppeteer-checks/src/get-url-slugs.js

```diff
@@ -0,0 +1,33 @@
+async function getSlugs(url, page, browser) {
```

Comment:
I'd like to discuss this logic in more detail when we meet.

We don't necessarily need to change anything in this PR, but I'd like to take the time to review how this works along with an example.

Typically, web crawling like this requires some kind of recursive logic like "visit every page to check links, then for every link found visit that page to check links...", where you "bottom out" once you've build the entire tree of pages starting at the root URL.

On the other hand, your current approach might get most of the possible links with only a single page request, which would involve significantly less network traffic.

Reply:
In the spirit of minimizing complexity, I'm inclined to say leave it for now and we can refactor if we find a significant fraction of links are being missed without the recursive approach.